src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,68 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import lotto.domain.dto.LottoDto;
+
+public class Lotto {
+
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = sortNumbers(numbers);
+    }
+
+    public int countSameNumber(List<Integer> candidateNumbers) {
+        int count = 0;
+
+        for (Integer candidate : candidateNumbers) {
+            if (numbers.contains(candidate)) {
+                count++;
+            }
+        }
+
+        return count;
+    }
```

Comment:
```suggestion
    public int countSameNumber(List<Integer> candidateNumbers) {
        return Math.toIntExact(candidateNumbers.stream()
                .filter(this::contains)
                .count());
    }
```
이렇게 리팩토링 해볼수 있을것 같아요